test/init_test.go

```diff
@@ -50,7 +51,27 @@ func init() {
 	flag.BoolVar(&skipRootUserTests, "skipRootUserTests", false, "Skip tests that require root user")
 }
 
+// excludeForArchitecture checks if name of calling function(name of the test) is in in the excluded list
+func excludeForArchitecture(callerLevel int) bool {
+	//get the Caller and the func name after
+	pc, _, _, _ := runtime.Caller(callerLevel)
```

Comment:
Could we use t.Name() instead? https://golang.org/src/testing/testing.go?s=22385:22415#L639

(I'm not actually sure, but it looks like it returns the current test name)

Reply:
Thank you so much! yes, you are right, t.Name() solves all my challenges with tests and subtests. Don't know how I lost it in my mind.. :(